rest-api/resources/src/main/java/org/eclipse/kapua/app/api/resources/v1/resources/DeviceManagementOperations.java

```diff
@@ -117,10 +122,6 @@ public DeviceManagementOperationListResult query(
             DeviceManagementOperationQuery query) throws KapuaException {
         query.setScopeId(scopeId);
 
-        AndPredicate andPredicate = query.andPredicate();
```

Comment:
This shouldn't be removed otherwise invoking directly `_query` will not filter by `deviceId`.
It should be something like:
```
if (query.getPredicate() != null) {
    AndPredicate andPredicate = query.andPredicate(
        query.attributePredicate(DeviceManagementOperationAttributes.DEVICE_ID, deviceId),
        query.getPredicate()
    )
    query.setPredicate(andPredicate);
}
else {
    query.setPredicate(query.attributePredicate(DeviceManagementOperationAttributes.DEVICE_ID, deviceId));
}
```
which always applies `deviceId` filtering and preserves `query.predicate` if specified (either from REST body or via  `simpleQuery`)

Reply:
Thanks for your feedback
That sounds good, and I've updated the code accordingly.